Updating the Slider Widget to allow up and down arrow keys to navigate out of the slider.

egramond opened the pull request:
Updating Slider Widget to allow up and down arrow keys to navigate out of the slider on directional navigation mode.

This prevents users using directional navigation from becoming stuck on the slider once they navigate to it.

gspencergoog commented on packages/flutter/lib/src/material/slider.dart:
```diff
@@ -474,13 +474,22 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
   // Keyboard mapping for a focused slider.
-  final Map<ShortcutActivator, Intent> _shortcutMap = const <ShortcutActivator, Intent>{
+  final Map<ShortcutActivator, Intent> _traditionalNavShortcutMap = const <ShortcutActivator, Intent>{
       SingleActivator(LogicalKeyboardKey.arrowUp): _AdjustSliderIntent.up(),
```
Are we sure that these are even correct.  I mean, it's convenient to have up and down change the slider, but are they actually idioms that anyone knows about?  I can see it being important for a vertical slider, but we don't have those.

Because if they're not really idiomatic, then we can just eliminate them and directional nav will just work.

egramond replied:
I don't know the history behind that particular bit of functionality so I went with the solution affected the least amount of things.  I'd personally be fine with that smaller change if you think it's safe.

egramond commented:
cc @clocksmith who initially added keyboard shortcut support to the Slider Widget and may know if the up and down shortcuts are idiomatic for some use cases.

engine-flutter-autoroll committed:
Roll Engine from 586c90c1336f to 4228b72286ff (4 revisions) (#102290)

engine-flutter-autoroll committed:
Roll Plugins from 6e18c7195678 to 5c3c1b8aac06 (7 revisions) (#102306)

yusuf-goog committed:
Switch ci.yaml Linux builders to Debian. (#102253)

In preparation for upgrading linux VMs to Ubuntu, specify builders
to use 'os: Debian' instead of 'os: Linux'.

There should be no functional changes to current builders with this change.

Bug:102252

TahaTesser committed:
Improve 'NestedScrollView and internal scrolling' test to account for all the inner children layers (#102309)

fluttergithubbot committed:
Marks Linux_android gradient_dynamic_perf__e2e_summary to be unflaky (#101832)

fluttergithubbot committed:
Marks Mac_ios cubic_bezier_perf_ios_sksl_warmup__timeline_summary to be flaky (#102231)

engine-flutter-autoroll committed:
Roll Engine from 4228b72286ff to 701e5e8f44ac (4 revisions) (#102316)

drewroengoogle committed:
Change dependency from ios_signing to apple_signing (#102259)

* Change dependency name to apple_signing for a better understanding of it's use

* Updated Mac_ios test to run if .ci.yaml is updated

chunhtai committed:
Adds tooltip to semantics node (#87684)

Abhishek01039 committed:
macos support for image_list (#84899)

Commit:
[ci.yaml] Migrate bringup targets to cocoon scheduler (#102249)

TahaTesser committed:
Replace `ListTile` with `Chip` in the `debugCheckHasMaterial control test` (#102311)

engine-flutter-autoroll committed:
Roll Engine from 701e5e8f44ac to 1cb7af579661 (1 revision) (#102322)

LongCatIsLooong committed:
Put `waitFor` back in keyboard_textfield_test integration test (#102315)

keyonghan committed:
Enable DeviceLab arm64 ios tests as `bringup: true` (#102089)

engine-flutter-autoroll committed:
Roll Engine from 1cb7af579661 to 4f05631a710a (1 revision) (#102326)

Commit:
Remove build_aar_module_test from devicelab tests (#102329)

jason-simmons committed:
Fix the code point for CupertinoIcons.location (#101923)

jmagman committed:
Handle CocoaPods ffi stderr (#102327)

iskakaushik committed:
Enable jank metric collection in profile mode (#102334)

dependabot[bot] committed:
Bump actions/checkout from 3.0.1 to 3.0.2 (#102337)

engine-flutter-autoroll committed:
Roll Engine from 4f05631a710a to 238eb9d7f55f (5 revisions) (#102340)

engine-flutter-autoroll committed:
Roll Engine from 238eb9d7f55f to 1b10b146cf7c (2 revisions) (#102344)

engine-flutter-autoroll committed:
Roll Engine from 1b10b146cf7c to bcd35000ed99 (1 revision) (#102345)

engine-flutter-autoroll committed:
Roll Engine from bcd35000ed99 to c6ea046c2cda (3 revisions) (#102347)

engine-flutter-autoroll committed:
Roll Engine from c6ea046c2cda to d4158760408d (1 revision) (#102349)

engine-flutter-autoroll committed:
Roll Engine from d4158760408d to e96c75676ee5 (2 revisions) (#102354)

engine-flutter-autoroll committed:
Roll Engine from e96c75676ee5 to 2d7fcf4e36eb (1 revision) (#102359)

engine-flutter-autoroll committed:
Roll Engine from 2d7fcf4e36eb to 0a88671c35f6 (2 revisions) (#102371)

engine-flutter-autoroll committed:
Roll Engine from 0a88671c35f6 to d9b3a550a75a (1 revision) (#102375)

engine-flutter-autoroll committed:
Roll Plugins from 5c3c1b8aac06 to ca63d964d9fd (10 revisions) (#102379)

engine-flutter-autoroll committed:
Roll Engine from d9b3a550a75a to 587bca0454bc (1 revision) (#102380)

engine-flutter-autoroll committed:
Roll Engine from 587bca0454bc to 4350d117406d (1 revision) (#102386)

engine-flutter-autoroll committed:
Roll Engine from 4350d117406d to ad6976c4d1db (1 revision) (#102394)

elliette committed:
Hide unresolved DartUri log messages (#102338)

engine-flutter-autoroll committed:
Roll Engine from ad6976c4d1db to 061fb1d74a9f (1 revision) (#102397)

engine-flutter-autoroll committed:
Roll Engine from 061fb1d74a9f to 0da451d25cf8 (1 revision) (#102400)

rgevrek committed:
Fix jumping when doing long press for selecting text (#102270)

DanielCardonaRojas committed:
Add matcher to find at least a given number of widgets (#102081) (#102342)

christopherfujino committed:
[flutter_tools] Add test that "Unresolve URI" warnings from DWDS are filtered from logs (#102399)

engine-flutter-autoroll committed:
Roll Engine from 0da451d25cf8 to 7cb80a7964ef (2 revisions) (#102409)

engine-flutter-autoroll committed:
Roll Engine from 7cb80a7964ef to 819ca4ec8cd5 (3 revisions) (#102411)

engine-flutter-autoroll committed:
Roll Engine from 819ca4ec8cd5 to e9ebbb5575fb (4 revisions) (#102425)

engine-flutter-autoroll committed:
Roll Engine from e9ebbb5575fb to 49fe29e373d6 (2 revisions) (#102428)

engine-flutter-autoroll committed:
Roll Engine from 49fe29e373d6 to f6725544f45e (1 revision) (#102437)

engine-flutter-autoroll committed:
Roll Engine from f6725544f45e to b4cc382cb26b (1 revision) (#102438)

engine-flutter-autoroll committed:
Roll Engine from b4cc382cb26b to dbe5a764d952 (1 revision) (#102444)

engine-flutter-autoroll committed:
Roll Engine from dbe5a764d952 to cf3445791f6d (1 revision) (#102448)

engine-flutter-autoroll committed:
Roll Engine from cf3445791f6d to f23d31c4755e (1 revision) (#102456)

engine-flutter-autoroll committed:
Roll Engine from f23d31c4755e to 0a924011e2f0 (1 revision) (#102461)

engine-flutter-autoroll committed:
Roll Engine from 0a924011e2f0 to 984b5b332bb2 (1 revision) (#102463)

AlexV525 committed:
[gen_l10n] Throw a specified exception when the language code is invalid (#102431)

engine-flutter-autoroll committed:
Roll Engine from 984b5b332bb2 to e31251e55fd9 (1 revision) (#102470)

engine-flutter-autoroll committed:
Roll Engine from e31251e55fd9 to 3200b99d99f3 (1 revision) (#102475)

engine-flutter-autoroll committed:
Roll Engine from 3200b99d99f3 to 20b5b8604c3e (1 revision) (#102477)

engine-flutter-autoroll committed:
Roll Engine from 20b5b8604c3e to 93de17f347fe (1 revision) (#102480)

engine-flutter-autoroll committed:
Roll Engine from 93de17f347fe to 10efa195bf1c (2 revisions) (#102488)

engine-flutter-autoroll committed:
Roll Engine from 10efa195bf1c to c901936d4f6c (1 revision) (#102493)

engine-flutter-autoroll committed:
Roll Engine from c901936d4f6c to 5451e88a09c0 (1 revision) (#102509)

engine-flutter-autoroll committed:
Roll Plugins from ca63d964d9fd to 4ff0157340ea (3 revisions) (#102511)

engine-flutter-autoroll committed:
Roll Engine from 5451e88a09c0 to 966812846313 (2 revisions) (#102513)

wangying3426 committed:
Assign debugCreator for more layers (#101899)

Commit:
Reduce Gradle log level in verbose output (#102422)

HelioStrike committed:
Enable TestDefaultBinaryMessenger to intercept all platform channels. (#100049)

engine-flutter-autoroll committed:
Roll Engine from 966812846313 to c4184728a9ad (1 revision) (#102515)

engine-flutter-autoroll committed:
Roll Engine from c4184728a9ad to f1bb8f421135 (1 revision) (#102519)

engine-flutter-autoroll committed:
Roll Engine from f1bb8f421135 to 0f47c7a68811 (1 revision) (#102523)

Piinks committed:
Re-land reverse case for AppBar scrolled under (#102343)

engine-flutter-autoroll committed:
Roll Engine from 0f47c7a68811 to 5fdfbe3010bb (1 revision) (#102526)

dependabot[bot] committed:
Bump debian from bullseye-20220328-slim to bullseye-20220418-slim in /dev/ci/docker_linux (#102528)

gaaclarke committed:
Made Directionality forego dependency tracking for better performance. (#102336)

engine-flutter-autoroll committed:
Roll Engine from 5fdfbe3010bb to 4df9cf27052b (1 revision) (#102529)

keyonghan committed:
Enable M1 mac/ios tests in prod as bringup: false (#102530)

Commit:
[framework] allow disabling image filter layer (#102085)

Commit:
add benchmark for animated complex opacity (#102532)

xu-baolin committed:
Fix a `Slider` widget accessibility bug (#102129)

Commit:
add missing benchmark test file fromhttps://github.com/flutter/flutter/pull/102532 (#102544)

engine-flutter-autoroll committed:
Roll Engine from 4df9cf27052b to 79152137a94c (8 revisions) (#102551)

* b2ca1ce53 [web] Migrate Flutter Web to JS static interop - 14. (flutter/engine#32683)

* d3886d6c5 Roll Fuchsia Linux SDK from MbA1RtjV4... to YyokSAVV4... (flutter/engine#32907)

* 0aca7c0a1 Roll Skia from 4f33f5bef8eb to f8e52b27aa70 (4 revisions) (flutter/engine#32911)

* ccf075470 Roll Skia from f8e52b27aa70 to 74742130b08d (2 revisions) (flutter/engine#32913)

* a2d8df99e Roll Skia from 74742130b08d to 82146a7fcde2 (1 revision) (flutter/engine#32914)

* 1e19613db Roll Fuchsia Mac SDK from JVNKiIJWP... to Jki3kG8_P... (flutter/engine#32916)

* 65bc7e348 Roll Dart SDK from 865c2ff0940e to e89b629ed91d (1 revision) (flutter/engine#32917)

* 79152137a Revert "[web] Migrate Flutter Web to JS static interop - 14. (#32683)" (flutter/engine#32919)

* Add impellerc to conductor codesigning list

* Update coductor codesigning for libtessellator.dylib

Co-authored-by: Zachary Anderson <zanderso@users.noreply.github.com>

engine-flutter-autoroll committed:
Roll Engine from 79152137a94c to df21738974e3 (1 revision) (#102553)

engine-flutter-autoroll committed:
Roll Engine from df21738974e3 to 8aace64d19fc (1 revision) (#102555)

engine-flutter-autoroll committed:
Roll Engine from 8aace64d19fc to cd78221a88ae (1 revision) (#102562)

bleroux committed:
Fix DraggableScrollableController.animateTo leaks Ticker (#102504)

dnfield committed:
Avoid scheduling a forced frame when there is no child to the renderView (#102556)

engine-flutter-autoroll committed:
Roll Engine from cd78221a88ae to 6d60b810d747 (1 revision) (#102575)

engine-flutter-autoroll committed:
Roll Plugins from 4ff0157340ea to e55659835427 (1 revision) (#102576)

Commit:
[framework] allow other RenderObjects to behave like repaint boundaries (#101952)

engine-flutter-autoroll committed:
Roll Engine from 6d60b810d747 to d5b381c2e56d (1 revision) (#102581)

Piinks committed:
Revert "Re-land reverse case for AppBar scrolled under" (#102580)

Commit:
[flutter_tools] remove UWP tooling (#102174)

engine-flutter-autoroll committed:
Roll Engine from d5b381c2e56d to 3f27923c7e6c (2 revisions) (#102588)

Commit:
[framework] Reland: use ImageFilter for zoom page transition  (#102552)

engine-flutter-autoroll committed:
Roll Engine from 3f27923c7e6c to b30b65a74554 (1 revision) (#102594)

jason-simmons committed:
Improve efficiency of copying the animation ObserverList in notifyListeners (#102536)

zanderso committed:
[flutter_tools] Add entries to HostArtifacts for impellerc, libtessellator (#102593)

keyonghan committed:
Enable Mac/android arm64 tests as `bringup: true` (#102589)

goderbauer committed:
Fix docs re: return value of Navigator's restorable methods (#102595)

engine-flutter-autoroll committed:
Roll Engine from b30b65a74554 to 545a0b4ad107 (4 revisions) (#102599)

engine-flutter-autoroll committed:
Roll Engine from 545a0b4ad107 to 07584c64bb7b (3 revisions) (#102603)

Commit:
[framework] remove usage and deprecate physical model layer (#102274)

engine-flutter-autoroll committed:
Roll Engine from 07584c64bb7b to 46b0889d13ce (1 revision) (#102606)

Commit:
Revert "[framework] Reland: use ImageFilter for zoom page transition " (#102611)

engine-flutter-autoroll committed:
Roll Engine from 46b0889d13ce to 6ac401970df1 (3 revisions) (#102612)

keyonghan committed:
Update open sdk version to be version `version:11` (#102610)

engine-flutter-autoroll committed:
Roll Engine from 6ac401970df1 to 24471f3cbb33 (3 revisions) (#102618)

engine-flutter-autoroll committed:
Roll Engine from 24471f3cbb33 to 5d844aa30fc6 (1 revision) (#102619)

engine-flutter-autoroll committed:
Roll Engine from 5d844aa30fc6 to fcc15947754b (1 revision) (#102622)

a14n committed:
add missing trailing commas in list/set/map literals (#102585)

engine-flutter-autoroll committed:
Roll Engine from fcc15947754b to 189cdc7df7ae (1 revision) (#102623)

engine-flutter-autoroll committed:
Roll Engine from 189cdc7df7ae to c5bb99941547 (4 revisions) (#102634)

JsouLiang committed:
Benchmark cannot run in Android12 phone (#100761)

zanderso committed:
Revert "Benchmark cannot run in Android12 phone (#100761)" (#102649)

This reverts commit 3d148821ebfda5c6c8ad74ca43c01fbb20c3415b.

engine-flutter-autoroll committed:
Roll Engine from c5bb99941547 to 6abf69920824 (2 revisions) (#102646)

engine-flutter-autoroll committed:
Roll Plugins from e55659835427 to 3e43f590d4d0 (3 revisions) (#102656)

pq committed:
rename local functions with `_`s (#102615)

guidezpl committed:
Fixes documentation inconsistencies around 'material' and 'material design' (#102632)

engine-flutter-autoroll committed:
Roll Engine from 6abf69920824 to 8f7e55b13064 (1 revision) (#102666)

engine-flutter-autoroll committed:
Roll Engine from 8f7e55b13064 to 9e6911e4da21 (2 revisions) (#102673)

Commit:
[intl] speed up localization generation and regenerate symbols (#102614)

fluttergithubbot committed:
Marks Mac tool_host_cross_arch_tests to be unflaky (#102659)

TahaTesser committed:
[reland] Refactor ToggleButtons (remove RawMaterialButton) (#101760)

darrenaustin committed:
Migrate AppBar to Material 3 (#101884)

engine-flutter-autoroll committed:
Roll Engine from 9e6911e4da21 to 94824fcb600a (6 revisions) (#102681)

fluttergithubbot committed:
Marks Linux_android android_semantics_integration_test to be unflaky (#102658)

Commit:
[ci.yaml] Migrate 50 targets to Cocoon scheduler (#102676)

engine-flutter-autoroll committed:
Roll Engine from 94824fcb600a to 35e56175dfde (1 revision) (#102686)

Piinks committed:
Fix AppBar scrolledUnder initial state - the third (#102582)

dependabot[bot] committed:
Bump github/codeql-action from 2.1.8 to 2.1.9 (#102693)

engine-flutter-autoroll committed:
Roll Engine from 35e56175dfde to 0ac2fc2a1d1c (2 revisions) (#102695)

pq committed:
[flutter_driver] rename local functions with `_`s (#102689)

pq committed:
[flutter_tools] rename local functions with `_`s (#102688)

TahaTesser committed:
Fix `BottomNavigationBar` label style text colors (#102638)

engine-flutter-autoroll committed:
Roll Engine from 0ac2fc2a1d1c to 561f96247933 (3 revisions) (#102700)

GaryQian committed:
Remove Linux android views (#102699)

engine-flutter-autoroll committed:
Roll Engine from 561f96247933 to b75aaa34f16d (2 revisions) (#102701)

engine-flutter-autoroll committed:
Roll Engine from b75aaa34f16d to 80c04d70c0d6 (435 revisions) (#102704)

* 28af36b9d Merge entity renderer impl into the entity renderer.

* 08f00cd5e Move entity shaders to their own directory.

* e95252f06 Shorten names of the render target attachment descriptors.

* b5ed864d7 Fixup names of pipeline descriptors.

* 531162ff5 Minor fixups to pipeline creation.

* a09363684 Make pipelines store pipeline descriptors.

* 612b54fd7 Start consolidating content rendering.

* 27473322e Separate save layer calls into a canvas pass.

* 97455db0f Start wiring up the entity playground.

* 778fc113b Allow shader libraries to look at multiple shader dylibs.

* 5405b9a6f Implement texture mapping.

* 866045d5f WIP on stroke rendering.

* d35f10ab1 Insert additional padding at the end of the struct if the size of the struct does not satisfy the alignment requirements of all its members.

* 41ed619fc Fix struct padding calculations.

* ac10b798e Rendering of continuous strokes.

* ede764556 Cleanup entity API.

* 9ce9a81d5 WIP Add test for clip rendering.

* 05705810a Add support for typed commands.

* 9c88b09aa Give the default render target a stencil buffer.

* 60c3b2191 Implement stencil-only clips.

* 65ffd0482 Rework canvas passes in prep for subpasses.

* 38d4abac5 Renderer components no longer need a surface reference to function.

* 17d301b96 WIP on creating utils for creating offscreen render targets.

* 935931173 Fix bug in texture usage mask.

* 3f417887c Setup aiks for canvas subpasses.

* ab16b0bc8 Setup canvas pass hierarchies.

* 3a8fb57fc Canvas subpasses rendering into their own command buffers.

* 398b74fac Work towards making canvas subpasses configure postprocessing effects.

* bb633cade Fix alpha writes while rendering to texture.

* f9a6a2cb1 Fixup ortho calculations and add ostream printers for geometry utils.

* 18f8e729b Setup a framework to collapse passes.

* d00efdd5f Move entity pass management to the entity framework.

* ead8058c9 Setup a paint pass delgate.

* ed8b78a96 Allow entity passes to be elided based on their contents.

* 026dcd191 Turn on full-screen 4xMSAA by default and additional render target validations.

* 603b91575 Add render pass and pipeline sample count validation.

* 30f28f161 All creation of pipeline variants from a pipeline.

* 2b7848053 Allow for the on-demand creation and caching of pipeline variants from the prototype.

* 723ca3b97 Add more validation logs instead of FML logs.

* 6d871081a Implement Canvas::Skew.

* 18399a5ff Implement Canvas::SaveLayer with bounds.

* 4234b343d Minor: Add Canvas::Transform.

* 8e90d388d Rect union and intersection.

* 4417fa637 Don't let subpass coverage exceed entity coverage.

* a63c04785 Remove unused and confusing API in TRect.

* 20d22fd3e Add macOS availability checks.

* 05e9f893a Add missing availability check for macOS 10.12.

* cb547949c Fix broken test.

* e3d435755 Support non-uniform rounding radii.

* 10487e3c1 Stub out a display list dispatcher.

* f83a0211a Support path fill types.

* 66096b5ba Add support for path composition in path builder.

* d3626cd1f Implement conversion SkPath to Impeller paths.

* 98c18b3e1 Import the archivist framework.

* 147e64fbd Update translation units for style guide.

* 8d7d8e653 Cleanup archive structure.

* 89589cc3d Separate archive_location into its own TU.

* 547fc005e Clearer names for primary keys.

* 4a6e2d07f Add uncommitted patches from a previous change.

* 0c595a929 More use of optionals in archivist.

* 52891357c Make primary keys optional.

* 7b3ab4178 Minor: Rename primary key variable name.

* 188890c35 Support user readable table names.

* 94b9bfd7c Add error message for incorrect member lookup.

* 95386ed3f Remove superfluous auto_key from ArchiveDef.

* bf24ff925 Cleanup macros in archive statement.

* 85b8a0093 Remove remaining C-Style casts.

* 4ca5af44c Fix temporary database file deletion.

* 84eee265a Fix iOS compilation issues.

* a104fa53d Update GN rules for Flutter updates.

* a404064ac Minor: Rename the Impeller dispatcher to be something sane and prep for sidecar wireup.

* 57a0728c1 Rework the confusingly named renderer subsystem instances.

* db3c4a11d Make the Metal drawable wrapping routine from the playground a utility.

* 7566d670c Minor: Change the names of the intermediates for the wrapped render target.

* 0016e4808 Allow resizing the playground.

* 64e3b29ee Holy size savings Batman!

* 3ac38d8b1 Forward targets specification to the code generator in impellerc.

* 0b63e2cc2 Shader data can now be compiled into the binary.

* 86c91dc62 Fix pipeline library issue related to not tracking pending futures.

* 687f18991 Don't waste space on an unnecessary depth buffer.

* 8748c7687 Setup default pixel formats.

* f282f414b Remove support for the weird 5 byte pixel format.

* 7e5364a4d Minor: Account for macro namespacing.

* c66b7fb34 IOS macros have not been namespaced yet.

* f70087318 Gate the minimum iOS deployment versions in the generated shaders.

* 819b0336f Cross reference issues and add additional tracing.

* 169abed1c Add a README.

* c251e079a Fixup minor Markdown formatting issues in the README.

* 6d520d234 Add an image overview of the shader compilation pipeline.

* a449707c0 Link to the shader pipeline from the README.

* 844d40063 Enable playgrounds in all tests and fix default sample counts.

* 9c536958c Update tests to account for sample count defaults.

* 14bf82761 Back out changes to the Impeller dispatcher stub that depend on WIP interface changes. (flutter/engine#3)

* 2a6bb6eb3 create scorecards-analysis.yml (flutter/engine#4)

* fec03558c Use -M0 instead of -frecord-sources (flutter/engine#5)

* e982526e8 Compensate for change to saveLayer (flutter/engine#6)

* 14ec50ada Adds a GN flag for playgrounds (flutter/engine#7)

* 3d422b2f7 Wrap all MD files at 80 cols, minor cleanup of headers/indenting (flutter/engine#8)

* ca3433db3 Fix Rect::Compare bugs (flutter/engine#10)

* e768f4d0a adopt new DlColorFilter objects (flutter/engine#12)

* 88f4c24ac Support 16bit index buffers (flutter/engine#11)

* dde655fdd Add viewport and scissor support (flutter/engine#9)

* 4220ef372 Add cull mode to command (flutter/engine#13)

* 9b2cbc13c Support nested clips & clip state restoration (flutter/engine#14)

* a2f755518 Fix bug introduced in a476b92 which broke a bunch of tests (flutter/engine#15)

* e7f90736b changes for new DlMaskFilter objects (flutter/engine#17)

* ba1ecb66c Add baseVertex (flutter/engine#16)

* 542b8c4c3 Fix index size check to respect 16 bit buffers (flutter/engine#19)

* 26090e475 Add mixed type specializations on algebraic ops in TPoint (flutter/engine#21)

* cb9997ab9 Respect TextureContents/Canvas::DrawImageRect source rect (flutter/engine#22)

* 9ecbb76c5 Add an ImGui backend targetting Impeller to the playground (flutter/engine#20)

* 604a93f96 Fix prod build (flutter/engine#23)

* 6cd7a4553 Bad cubic tests (flutter/engine#24)

* e94acaae6 Match skia's path param ordering (flutter/engine#25)

* b26c6ac38 Do not include the leading point for cubic polylines (flutter/engine#29)

* bafa73b54 Add assignment operators to point (flutter/engine#26)

* 2373da89f Add entity playground callback (flutter/engine#27)

* 2dceced0e adopt new DlAttributeType enum naming (flutter/engine#30)

* f41c0a93e Support subcontours in strokes, lay groundwork for fills (flutter/engine#31)

* 4be3053ec Add immediate mode manipulator widget macros for the playground (flutter/engine#28)

* f0a7b4e90 Fix MoveTo for fills (flutter/engine#32)

* df2485c90 Add dot and cross product to Point (flutter/engine#33)

* 3cff1a93b Remove duplicate points between connected components when generating polylines (flutter/engine#34)

* 451f93e83 Handle all corner cases for stroke geometry, add bevel join & cap/join enums (flutter/engine#35)

* 0dae5ada0 Add `//impeller/typographer`. (flutter/engine#36)

* e1a9ae853 Remove schedule runs of scorecards. (flutter/engine#38)

* 8d79ce674 Bump ossf/scorecard-action from 1.0.3 to 1.0.4 (flutter/engine#40)

* 137fa4816 Fix license to make the license script happy (flutter/engine#37)

* 91ec4c207 Bump github/codeql-action from 1.0.26 to 1.1.3 (flutter/engine#39)

* 62135be77 Remove break corner cases, simplify strokes, and generate closed path joins (flutter/engine#41)

* 9f9c557ec Account for glyph extents in the atlas. (flutter/engine#42)

* 4e323c6fd Relay text color from Paint. (flutter/engine#43)

* c11f4083d Account for updated macro name in //flutter/fml/build_config.h (flutter/engine#44)

* a781e2f9a Setup join/cap proc (flutter/engine#45)

* 555fabaa0 Avoid depending on STB in //flutter/impeller/image. (flutter/engine#47)

* d84a831d8 Add square cap (flutter/engine#48)

* 75e83ff3e Point: Add reflect and make scalar ops more flexible (flutter/engine#51)

* d9e792783 Add miter join (flutter/engine#49)

* 905895695 Add round caps and joins (flutter/engine#52)

* c81fbbdac Move Contents classes into separate translation units (flutter/engine#53)

* 9d6048ab0 Expose C API for tessellator, move to own subdir (flutter/engine#18)

* 2a1b56096 Fix ScalarNearlyEqual test (flutter/engine#56)

* 845c3bc07 Add color premultiply/unpremultiply (flutter/engine#57)

* 6f9fa60b0 Add pipeline blend modes & demo (flutter/engine#55)

* 0936dfaee Bump github/codeql-action from 1.1.3 to 1.1.4 (flutter/engine#58)

* 6da61e7fd Remove FML dependency on geometry, tessellator (flutter/engine#59)

* b14fb96b3 Make Path::Polyline::GetContourPointBounds safe for OOB (flutter/engine#60)

* 24d7083c9 Add LICENSE file. (flutter/engine#61)

* 841bb991b Move impeller::Comparable to //impeller/base. (flutter/engine#62)

* 74dff11d5 Add a script that checks that all source files have a valid license block. (flutter/engine#63)

* ec27727a2 Correct polyline bounds test (flutter/engine#64)

* 0084668ec Bump actions/upload-artifact from 2.3.1 to 3 (flutter/engine#54)

* b816f8f4f Bump actions/checkout from 2.4.0 to 3 (flutter/engine#50)

* e5fc54887 [tessellator]Expose and document smoothing approximation via the C/Dart API. Minor refactor of tessellator to supply fill type to method rather than to object. (flutter/engine#65)

* 24e249a4f Fix top-level build group to not include the compiler and fixup headers. (flutter/engine#66)

* 9c3cbf855 Hack to prevent back-to-back playground tests from hanging (flutter/engine#68)

* 1828c4c89 Correct default blend mode, fix solid stroke shader to respect premultiplied source colors (flutter/engine#69)

* b3cbf05b8 Chainable texture filters (flutter/engine#67)

* f3a6b3e1b Document GN rules and make targets that cannot be built on the platform be no-ops. (flutter/engine#70)

* 597c10eb1 Fix diagnostic on iOS simulators. (flutter/engine#71)

* e46345db6 Only depend on FML for tracing when building executables. (flutter/engine#72)

* d0c59286c Fix loop-range-construct warnings added in newer versions of Clang. (flutter/engine#75)

* b97e965b7 Add remaining additive blends (flutter/engine#76)

* ec1dd96b0 Fix issues with constexpr correctness. (flutter/engine#77)

* 47cb97c35 Add case necessary on 32-bit arm builds. (flutter/engine#79)

* f5db75a15 Use new DlColorSource objects (flutter/engine#73)

* 75a9853fe update the linear gradient code to the new accessor names (flutter/engine#80)

* e381974b1 Impelement DisplayListDispatcher::transformReset. (flutter/engine#82)

* 8784ff4be Don't use `Add[Thing]` operations when emulating Skia path components (flutter/engine#86)

* 9e33c6fc1 Transform clips by the entity's transform (flutter/engine#87)

* d916e2c7c Add blend filter support for advanced blends (flutter/engine#81)

* 396ef7d0a Bump github/codeql-action from 1.1.4 to 1.1.5 (flutter/engine#83)

* 503dff6b0 Add directional + 2D gaussian blur (flutter/engine#84)

* 6e778862f Add ClearContents to apply specified contents to the entire render target. (flutter/engine#88)

* 688321533 Add blend mode setting to Entity, Aiks paint, and the dispatcher (flutter/engine#89)

* 1d6dc1e68 Forward blend mode from the dispatchers drawColor call. (flutter/engine#90)

* 5b1b9dfbf Apply Aiks transforms in the canvas space, not world space (flutter/engine#91)

* b32a65fe9 Use new DlBlendMode object (flutter/engine#78)

* 47ff8dbed switch dispatcher methods to new DlImageFilter objects (flutter/engine#92)

* 3235eac2a Separate filters into different TUs (flutter/engine#93)

* f2d23512b Implement impeller::DisplayListDispatcher::drawTextBlob. (flutter/engine#94)

* a6b0939fc Add support for instanced rendering and shader storage buffers. (flutter/engine#95)

* 47e9b0853 Use 8bpp glyph atlases. (flutter/engine#96)

* 86c5ea4f3 Draw all glyphs in a text run using a single draw call. (flutter/engine#97)

* 32b5a8dd7 Batch all draw calls in the entire text frame. (flutter/engine#100)

* 39f4ebd75 Make the filter DAG render all textures at the correct resolution (flutter/engine#99)

* 0aaa84e5c Render glyphs at the correct resolution with respect to the current transform (flutter/engine#101)

* eb7b7d187 Add difference clipping (flutter/engine#104)

* 6f02b8943 Wire up stroke cap/join/miter limit display list ops (flutter/engine#105)

* cbd14baf1 Premultiply gradient colors (flutter/engine#106)

* 9886da891 Use display list images. (flutter/engine#102)

* 82a340332 Make filter inputs lazy and sharable (flutter/engine#103)

* 69334fd7b Implement drawArc (flutter/engine#109)

* 6de2c9c56 support new DlVertices object in dispatcher (flutter/engine#108)

* 31812de08 Bump github/codeql-action from 1.1.5 to 2.1.6 (flutter/engine#110)

* 78eb5d67f Add blur styles to gaussian blur (flutter/engine#107)

* c4f1a592d Fix solid stroke contour switching + round cap smoothing, and add transparent path overdraw example (flutter/engine#112)

* f2bf47d25 Fix SaveLayer/entity subpass behavioral problems (flutter/engine#111)

* 2dd379708 Wire up enhanced command buffer error reporting. (flutter/engine#113)

* 6e8368c47 Remove extra point from DrawRect; skip over duplicate contour points when generating polylines (flutter/engine#114)

* 93c59198b Explicitly wait for subpass buffers to be scheduled. (flutter/engine#115)

* f2e297c1f Bump github/codeql-action from 2.1.6 to 2.1.7 (flutter/engine#116)

* e09b0ed12 Add static thread safety analysis ready synchronization primitives. (flutter/engine#117)

* 8075d383f Standardize around "coverage" terminology; make coverage optional everywhere (flutter/engine#118)

* 685ae23ad Add explicit scaling methods for Point/Vector2 (flutter/engine#119)

* af922f9d4 Ignore the stencil in the imgui pipeline (flutter/engine#120)

* 4248b5d2a Use the stencil buffer to avoid overdraw when rendering non-opaque solid strokes. (flutter/engine#121)

* 93035a4cd Remove base.h from base/BUILD (flutter/engine#122)

* 20148b270 Standardize around blur sigma<->radius conversion factor (flutter/engine#123)

* 533b86d3d No newline in depfile output, which seems to confuse ninja (flutter/engine#125)

* 0ed6d4c4a Fix Path::GetBoundingBox crash for cubics with no local min/max (flutter/engine#126)

* 1a055b08a Add solid stroke coverage override (flutter/engine#127)

* 3bc1f3340 Allow building targets that do not create metal shaders on all platforms. (flutter/engine#133)

* 43a750813 Move rendering TUs behind own flag (flutter/engine#134)

* 3c9f0e147 Build fixes for roll (flutter/engine#135)

* f6508c108 Remove extra premultiply from solid stroke/fill (flutter/engine#124)

* 999978fa9 Fix enum typedef for older mac SDKs (flutter/engine#136)

* 9c4018f10 Filters: Replace Snapshot position with a full transform (flutter/engine#130)

* ca3cd7f77 Move static methods to anonymous namespace (flutter/engine#137)

* f1c1b68c5 Adds --sksl target to impellerc (flutter/engine#131)

* fa68eb6eb Bump github/codeql-action from 2.1.7 to 2.1.8 (flutter/engine#128)

* a08cb8b4f Border mask blur (flutter/engine#132)

* a4599bf6f Add impellerc options to compile OpenGL Desktop and ES variant shaders. (flutter/engine#139)

* 8b881c634 Compute text coverage; use blend mode in savelayer; conservative pass collapse/elision behavior (flutter/engine#129)

* 3b1dfc4a4 Add an OpenGL ES stub and parameterize all playgrounds on rendering backend. (flutter/engine#141)

* 4d5031540 Fix subpass ordering (flutter/engine#143)

* 80bb5f953 Windows (flutter/engine#144)

* 79f2e86a4 Compile and package all shaders for the OpenGL ES backend. (flutter/engine#146)

* f86c9b72a Implement mask blur in display list dispatcher (flutter/engine#142)

* 01e81ef7e Fix Mac compilation issue. (flutter/engine#147)

* 2b48a800e Fix up build/test issues when building on or off of mac (flutter/engine#148)

* 68d88aeb4 Filters: Add local transforms (flutter/engine#140)

* df261f9d9 Fix minor mask blur issues (flutter/engine#149)

* c5474f6f0 Untie paths from entities (flutter/engine#145)

* 69b676c49 Bump actions/checkout from 3.0.0 to 3.0.2 (flutter/engine#151)

* 8679fe78e Remove dependency on Dart, add crash handler to impellerc (flutter/engine#150)

* 56056357f Use New DlPathEffect Object (flutter/engine#98)

* cd1a5a355 implement drawDisplayList in impeller dispatcher (flutter/engine#153)

* 5ec32d7ed Revert "Use New DlPathEffect Object (#98)" (flutter/engine#154)

* da9cfb3ca Separate FilterInput variants into their own TUs (flutter/engine#152)

* 3ebfe9e05 Another coverage fix for border mask blur (flutter/engine#158)

* 86fdc7912 Make it an error for a stage input to take more than one slot. (flutter/engine#161)

* 120c0b631 Avoid instancing/SSBOs for text to support GLES (flutter/engine#160)

* 4c75fe6e3 Set path for linear gradient contents; don't fail renders for failed tessellations (flutter/engine#162)

* dd4d7050e Check if sub_command_buffer is null before setting label (flutter/engine#163)

* d5f978195 format

* e676fd7a8 Missing files

* 89a30a729 temporarily increase clang_tidy test limits

* 7f0e95b1e Delete DartIsolateTest::ValidLoadingUnitSucceeds (flutter/engine#32970)

* 80c04d70c Delete AppLifecycleTests testDismissedFlutterViewControllerNotRespondingToApplicationLifecycle (flutter/engine#32971)

engine-flutter-autoroll committed:
Roll Engine from 80c04d70c0d6 to 220613fc0b20 (3 revisions) (#102708)

engine-flutter-autoroll committed:
Roll Engine from 220613fc0b20 to ba644fafb2c5 (1 revision) (#102715)

engine-flutter-autoroll committed:
Roll Engine from ba644fafb2c5 to 67940eb62023 (2 revisions) (#102717)

engine-flutter-autoroll committed:
Roll Engine from 67940eb62023 to b62ea5dce007 (1 revision) (#102721)

engine-flutter-autoroll committed:
Roll Engine from b62ea5dce007 to 7ffb3c560939 (1 revision) (#102729)

engine-flutter-autoroll committed:
Roll Engine from 7ffb3c560939 to 966757c6cc8e (1 revision) (#102740)

Piinks committed:
Revert "Fix `BottomNavigationBar` label style text colors" (#102756)

TahaTesser committed:
`ReorderableListView`: fix broken dartpad example & update examples, add tests  (#102723)

pq committed:
fix `unnecessary_null_checks` (#102705)

pq committed:
[dev, bots, examples] rename local functions with `_`s (#102703)

engine-flutter-autoroll committed:
Roll Plugins from 3e43f590d4d0 to ac3167f83402 (4 revisions) (#102768)

engine-flutter-autoroll committed:
Roll Engine from 966757c6cc8e to dc2a7da549d2 (1 revision) (#102770)

QuncCccccc committed:
Add padding for Navigation Bar to account for safe area (#102419)

apeltop committed:
Correct typos (#102487)

Commit:
Update features.dart (#102766)

chunhtai committed:
Add RouteInformationParser.parseRouteInformationWithDependencies (#102414)

engine-flutter-autoroll committed:
Roll Engine from dc2a7da549d2 to 649fe1080d56 (14 revisions) (#102795)

engine-flutter-autoroll committed:
Roll Engine from 649fe1080d56 to dfdfe0b3b042 (1 revision) (#102798)

engine-flutter-autoroll committed:
Roll Plugins from ac3167f83402 to 160c714e75f3 (9 revisions) (#102824)

* c09ef582e [flutter_plugin_tools] Remove UWP (flutter/plugins#5432)

* 3876f54bf [ci.yaml] Remove explicit caches (flutter/plugins#5434)

* 0605d876d [webview_flutter_android] Updates `pigeon` version to support null safety (flutter/plugins#5395)

* 4b687c9fa [video_player] Fix XCUITest based on the new tooltip accessibility label (flutter/plugins#5426)

* e777e515b Roll Flutter from 1b58a593deac to 4cea9afc1224 (147 revisions) (flutter/plugins#5437)

* 656e8c443 Roll Flutter from 4cea9afc1224 to 5b713147404d (4 revisions) (flutter/plugins#5441)

* f689280bf [flutter_plugin_tools] Validate code blocks in readme-check (flutter/plugins#5436)

* 9c41c6895 Roll Flutter from 5b713147404d to 7a74222832f8 (1 revision) (flutter/plugins#5442)

* 160c714e7 Roll Flutter from 7a74222832f8 to 2eed8cbf9370 (1 revision) (flutter/plugins#5443)

engine-flutter-autoroll committed:
Roll Engine from dfdfe0b3b042 to 311e36ac0297 (1 revision) (#102800)

yjbanov committed:
[web] allow small golden deltas in HTML renderer (#102791)

itsjustkevin committed:
update post release steps with links to release items (#102822)

prateekmedia committed:
Improvements to SearchDelegate (#91982)

TahaTesser committed:
`ToggleButtons`: Add interactive example (#100124)

TahaTesser committed:
Add `surfaceTintColor` to `NavigationBar` (#102628)

TahaTesser committed:
`Hero`: Add an example for `createRectTween` (#102650)

engine-flutter-autoroll committed:
Roll Engine from 311e36ac0297 to 2f7c922f407d (7 revisions) (#102834)

LongCatIsLooong committed:
Defer the OverlayEntry listenable disposal until its widget is unmounted (#102794)

engine-flutter-autoroll committed:
Roll Engine from 2f7c922f407d to 757ca7debdd9 (2 revisions) (#102839)

TahaTesser committed:
Update FAB M3 token template class to new naming scheme and fix a typo in `card.dart` (#102654)

Commit:
delete fast reassemble code (#102842)

engine-flutter-autoroll committed:
Roll Engine from 757ca7debdd9 to fbed11637817 (1 revision) (#102846)

zanderso committed:
Revert "delete fast reassemble code (#102842)" (#102856)

This reverts commit ec8693e80bf2c9f31c06d9832ebb4bccc5274440.

chunhtai committed:
Adds RouterConfig to simply API (#102786)

engine-flutter-autoroll committed:
Roll Engine from fbed11637817 to 097e58d1470e (6 revisions) (#102857)

godofredoc committed:
Add mirroring from master to main. (#102843)

engine-flutter-autoroll committed:
Roll Engine from 097e58d1470e to 14bb0c77bc66 (5 revisions) (#102859)

engine-flutter-autoroll committed:
Roll Engine from 14bb0c77bc66 to 433de86744b6 (1 revision) (#102860)

Jasguerrero committed:
[flutter_tool] New command project (#102118)

engine-flutter-autoroll committed:
Roll Engine from 433de86744b6 to 49c7db6429f9 (1 revision) (#102864)

engine-flutter-autoroll committed:
Roll Engine from 49c7db6429f9 to 863f646c7e1e (2 revisions) (#102867)

engine-flutter-autoroll committed:
Roll Engine from 863f646c7e1e to a6cd0c9398e8 (1 revision) (#102871)

engine-flutter-autoroll committed:
Roll Engine from a6cd0c9398e8 to 6b28127914fa (1 revision) (#102872)

engine-flutter-autoroll committed:
Roll Engine from 6b28127914fa to b7b880f327be (1 revision) (#102875)

engine-flutter-autoroll committed:
Roll Engine from b7b880f327be to 77c6a028058a (2 revisions) (#102880)

tvolkert committed:
Add widget of the week videos (#102862)

engine-flutter-autoroll committed:
Roll Engine from 77c6a028058a to b8d80cdaf230 (1 revision) (#102883)

engine-flutter-autoroll committed:
Roll Engine from b8d80cdaf230 to 4e3eabd30225 (1 revision) (#102884)

engine-flutter-autoroll committed:
Roll Engine from 4e3eabd30225 to 29a6bec85a3d (1 revision) (#102885)

engine-flutter-autoroll committed:
Roll Engine from 29a6bec85a3d to 9a1f497f1d90 (1 revision) (#102889)

engine-flutter-autoroll committed:
Roll Engine from 9a1f497f1d90 to 0accc792496e (1 revision) (#102891)

engine-flutter-autoroll committed:
Roll Engine from 0accc792496e to 1636f0064088 (1 revision) (#102892)

engine-flutter-autoroll committed:
Roll Engine from 1636f0064088 to d00bf2c6590e (1 revision) (#102896)

engine-flutter-autoroll committed:
Roll Engine from d00bf2c6590e to e59fa964dc78 (1 revision) (#102899)

engine-flutter-autoroll committed:
Roll Engine from e59fa964dc78 to 40ca11426a9f (1 revision) (#102900)

engine-flutter-autoroll committed:
Roll Engine from 40ca11426a9f to 2f43ad6044dd (1 revision) (#102901)

engine-flutter-autoroll committed:
Roll Engine from 2f43ad6044dd to bfa5f5df992a (1 revision) (#102905)

engine-flutter-autoroll committed:
Roll Engine from bfa5f5df992a to f57cdbd4680a (1 revision) (#102907)

engine-flutter-autoroll committed:
Roll Engine from f57cdbd4680a to ed074738efcc (2 revisions) (#102913)

engine-flutter-autoroll committed:
Roll Engine from ed074738efcc to b8e8fcb9b564 (1 revision) (#102917)

engine-flutter-autoroll committed:
Roll Engine from b8e8fcb9b564 to 66430ae6c18e (1 revision) (#102919)

engine-flutter-autoroll committed:
Roll Engine from 66430ae6c18e to 0c4573cd98da (1 revision) (#102922)

jmagman committed:
Embed Flutter and App frameworks for add-to-app on iOS (#102538)

engine-flutter-autoroll committed:
Roll Engine from 0c4573cd98da to 302393580add (1 revision) (#102926)

engine-flutter-autoroll committed:
Roll Plugins from 160c714e75f3 to bbcb88d3bae4 (20 revisions) (#102927)

TahaTesser committed:
Fix `CupertinoDatePicker`dark mode text color (#100312)

Improved code

TahaTesser committed:
Fix `CupertinoFormRow`dark mode text color (#100313)

Improved code

TahaTesser committed:
Fix `CupertinoPicker`dark mode text color (#100310)

Improved code

moffatman committed:
Fix position of CupertinoContextMenu within Transform.scale (#97896)

darrenaustin committed:
Revert "Fix position of CupertinoContextMenu within Transform.scale (#97896)" (#102938)

This reverts commit 6ddb99e98f8915f350f0da78d9b3efc4095b7c02.

keyonghan committed:
Deflake arm64 android tests (#102929)

engine-flutter-autoroll committed:
Roll Engine from 302393580add to cafc4eb02659 (4 revisions) (#102930)

christopherfujino committed:
[flutter_tools] update feature defaults (#102937)

engine-flutter-autoroll committed:
Roll Engine from cafc4eb02659 to 9018e487d9f8 (3 revisions) (#102941)

engine-flutter-autoroll committed:
Roll Engine from 9018e487d9f8 to 5a0432f959e1 (2 revisions) (#102949)

engine-flutter-autoroll committed:
Roll Engine from 5a0432f959e1 to a00140cb28fb (1 revision) (#102953)

engine-flutter-autoroll committed:
Roll Engine from a00140cb28fb to 53eeb5a1c7e0 (2 revisions) (#102956)

engine-flutter-autoroll committed:
Roll Engine from 53eeb5a1c7e0 to 6919a5114f01 (1 revision) (#102959)

engine-flutter-autoroll committed:
Roll Engine from 6919a5114f01 to 07819e5f45dd (1 revision) (#102961)

engine-flutter-autoroll committed:
Roll Engine from 07819e5f45dd to 5cb4391241dd (2 revisions) (#102965)

engine-flutter-autoroll committed:
Roll Engine from 5cb4391241dd to a3b3511fd773 (1 revision) (#102972)

engine-flutter-autoroll committed:
Roll Plugins from bbcb88d3bae4 to 60b0b2d31ee0 (14 revisions) (#102979)

engine-flutter-autoroll committed:
Roll Engine from a3b3511fd773 to 84e364559d4c (1 revision) (#102980)

TahaTesser committed:
Fix `CupertinoTimerPicker` dark mode text color (#100311)

pq committed:
[devicelab, flutter_test, ...] rename local functions with `_`s (#102833)

engine-flutter-autoroll committed:
Roll Engine from 84e364559d4c to 0a39dcddd3fd (1 revision) (#102982)

engine-flutter-autoroll committed:
Roll Engine from 0a39dcddd3fd to 87226dbd54ef (2 revisions) (#102984)

engine-flutter-autoroll committed:
Roll Engine from 87226dbd54ef to 37c3d89a78e0 (1 revision) (#102987)

Commit:
partial revert of repaint boundary change (#102962)

engine-flutter-autoroll committed:
Roll Engine from 37c3d89a78e0 to 6d9e19184d34 (4 revisions) (#102989)

Commit:
Clean up ci.yaml and add missing gradle caches (#102942)

fzyzcjy committed:
(Test-only) Add tests for new `ImageFilter.dilate`/`ImageFilter.erode` in flutter engine (#101036)

engine-flutter-autoroll committed:
Roll Engine from 6d9e19184d34 to acced3da4e72 (2 revisions) (#102999)

engine-flutter-autoroll committed:
Roll Engine from acced3da4e72 to a98987d78c8a (3 revisions) (#103007)

engine-flutter-autoroll committed:
Roll Engine from a98987d78c8a to a7f7ff020ddf (1 revision) (#103012)

engine-flutter-autoroll committed:
Roll Engine from a7f7ff020ddf to 0b46116471cc (1 revision) (#103017)

engine-flutter-autoroll committed:
Roll Engine from 0b46116471cc to 84d353906299 (2 revisions) (#103021)

moffatman committed:
Reland "Fix position of CupertinoContextMenu within Transform.scale" (#102943)

* Fix position of CupertinoContextMenu within Transform.scale (#97896)

* Fix after rebase

engine-flutter-autoroll committed:
Roll Engine from 84d353906299 to 0c5d7c579a42 (3 revisions) (#103026)

engine-flutter-autoroll committed:
Roll Engine from 0c5d7c579a42 to 95207488f999 (2 revisions) (#103027)

engine-flutter-autoroll committed:
Roll Engine from 95207488f999 to 6e49e558090c (1 revision) (#103030)

engine-flutter-autoroll committed:
Roll Engine from 6e49e558090c to e32059bed53e (1 revision) (#103033)

guidezpl committed:
Add prominent link to Google Fonts from icons documentation (#102808)

* Add prominent link to google fonts from icons

* Update icons.dart

* Update icons.dart

engine-flutter-autoroll committed:
Roll Plugins from 60b0b2d31ee0 to cafc154de894 (12 revisions) (#103051)

zanderso committed:
Don't hardcode ink sparkle spir-v (#102674)

engine-flutter-autoroll committed:
Roll Engine from e32059bed53e to d73708ca8882 (2 revisions) (#103056)

jeremyschlatter committed:
Fix crash from alt-tab'ing just after startup (#102453)

darrenaustin committed:
Revert "Fix crash from alt-tab'ing just after startup (#102453)" (#103074)

This reverts commit 74e9a6d9cd8a26287b14b88790a104ea41f9de65.

guidezpl committed:
Fix `NavigationDestination` API doc (#103031)

Commit:
[ci.yaml] Revert devicelab targets back to LUCI scheduler (#103048)

fluttergithubbot committed:
Marks Mac_ios external_ui_integration_test_ios to be flaky (#103058)

jmagman committed:
Update ffi gem installation instructions (#103008)

Piinks committed:
Add assertion that snackbars have a place to go (#103006)

fluttergithubbot committed:
Marks Mac_arm64_android run_release_test to be flaky (#103060)

jmagman committed:
Remove unused flutterMacOSPodspec artifact (#103088)

egramond commented:
ok so I screwed up this PR beyond repair with some git upstream command. I'll start a fresh one.

gspencergoog commented:
We've all been there. :-) Just let me know what the new one is and I'll review it.
